src/librbd/ImageWatcher.cc

```diff
@@ -1016,11 +1017,18 @@ void ImageWatcher<I>::handle_error(uint64_t handle, int err) {
   RWLock::WLocker l(m_watch_lock);
   if (m_watch_state == WATCH_STATE_REGISTERED) {
     m_watch_state = WATCH_STATE_ERROR;
+    m_watch_error_count += 1;
```

Comment:
Nit: this needs to be outside the ```if (m_watch_state == WATCH_STATE_REGISTERED)``` clause and then you can drop the ```else if``` clause

Reply:
Hmm. Just to double check: I decided against doing that because I wasn't sure what happens with that counter (maybe triggering unnecessary double rewatches) in the case UNREGISTERED case. That means we'd be accumulating errors there which will then cause a double rewatch later?

Reply:
... but if you reset the counter in ```::rewatch``` and avoid scheduling a re-watch if it's unregistered (or has an unregister pending), you should be safe.